packages/@glimmer/runtime/lib/lifetime.ts

```diff
@@ -1,18 +1,34 @@
-import { takeAssociated, snapshot, destructor, LINKED } from '@glimmer/util';
+import { takeAssociated, peekAssociated, snapshot, destructor, LINKED } from '@glimmer/util';
 import { DEBUG } from '@glimmer/local-debug-flags';
 import { clear } from './bounds';
 import { BlockOpcode } from './vm/update';
 import { Option, Bounds, Environment } from '@glimmer/interfaces';
 import { SimpleNode } from '@simple-dom/interface';
 
+export function legacySyncReset(parent: object, env: Environment) {
+  let linked = peekAssociated(parent);
+
+  if (linked !== null) {
+    env.willDestroy(snapshot(linked));
+  }
+}
+
 export function asyncReset(parent: object, env: Environment) {
   let linked = takeAssociated(parent);
 
-  if (linked) {
+  if (linked !== null) {
     env.didDestroy(snapshot(linked));
   }
 }
 
+export function legacySyncDestroy(parent: object, env: Environment) {
+  if (DEBUG) {
+    console.log('legacySyncDestroy', parent, LINKED.get(parent));
```

Comment:
Is this supposed to be left in? Seems like if we hit this a lot, Ember users would be annoyed... 🤔

Reply:
Chatted offline with @pzuraq, this `DEBUG` constant here is imported from local debug, so that isn't a large concern. Making another comment RE: changing that variable name...